handler/enqueue_annotation.go

```diff
@@ -30,21 +32,56 @@ import (
 var log = logf.Log.WithName("event_handler")
 
 const (
-	// NamespacedNameAnnotation - annotation that will be used to get the primary resource namespaced name.
+	// NamespacedNameAnnotation is an annotation whose value encodes the name and namespace of a resource to reconcile
+	// when a resource containing this annotation changes. Valid values are of the form `<namespace>/<name>` for
+	// namespace-scoped owners and `<name>` for cluster-scoped owners.
 	NamespacedNameAnnotation = "operator-sdk/primary-resource"
-	// TypeAnnotation - annotation that will be used to verify that the primary resource is the primary resource to use.
+	// TypeAnnotation is an annotation whose value encodes the group and kind of a resource to reconcil when a
+	// resource containing this annotation changes. Valid values are of the form `<Kind>` for resource in the
+	// core group, and `<Kind>.<group>` for all other resources.
 	TypeAnnotation = "operator-sdk/primary-resource-type"
 )
 
-// EnqueueRequestForAnnotation enqueues Requests based on the presence of an annotation that contains the
-// namespaced name of the primary resource.
+// EnqueueRequestForAnnotation enqueues Request containing the Name and Namespace specified in the
+// annotations of the object that is the source of the Event. The source of the event triggers reconciliation
+// of the parent resource which is identified by annotations. `NamespacedNameAnnotation` and `TypeAnnotation` together uniquely
+// identify an owner resource to reconcile.
+// handler.EnqueueRequestForAnnotation can be used to trigger reconciliation of resources which are cross-referenced.
+// By that we mean that a namespace-scoped dependent can trigger reconciliation of an owner which is not cluster-scoped, and
+// a cluster-scoped dependent can trigger the reconciliation of a namespace(scoped)-owner.
+
+// As an example, consider the case where we would like to watch clusterRole resource based on which we reconcile ReplicaSets which are
+// namespace scoped. With native owner references, this would not be possible since the cluster-scoped dependent (ClusterRole) is trying to
+// specify a namespace-scoped owner (ReplicaSet). Whereas in case of annotations-based handlers, we could implement the following:
```

Comment:
```suggestion
// specify a namespace-scoped owner (replicasets). Whereas in case of annotations-based handlers, we could implement the following:
```